tests/phpunit/tests/query/stickies.php

```diff
@@ -104,4 +105,30 @@ public function set_ignore_sticky_posts( $q ) {
 	public function set_post__not_in( $q ) {
 		$q->set( 'post__not_in', array( self::$posts[8] ) );
 	}
+
+	/**
+	 * @ticket 36907
+	 */
+	public function test_stickies_nest_query() {
+		add_filter( 'posts_pre_query', array( $this, 'get_query_from_filter' ), 10, 2 );
+		$this->go_to( '/' );
+
+		$q = $GLOBALS['wp_query'];
+		$this->assertNotEmpty( $this->sticky_query_vars['posts_per_page'] );
+		$this->assertSame( $q->query_vars['suppress_filters'], $this->sticky_query_vars['suppress_filters'] );
+		$this->assertSame( $q->query_vars['update_post_meta_cache'], $this->sticky_query_vars['update_post_meta_cache'] );
+		$this->assertSame( $q->query_vars['update_post_term_cache'], $this->sticky_query_vars['update_post_term_cache'] );
+		$this->assertSame( $q->query_vars['lazy_load_term_meta'], $this->sticky_query_vars['lazy_load_term_meta'] );
+		$this->assertSame( $q->query_vars['cache_results'], $this->sticky_query_vars['cache_results'] );
+		$this->assertTrue( $this->sticky_query_vars['ignore_sticky_posts'] );
+		$this->assertTrue( $this->sticky_query_vars['no_found_rows'] );
+	}
+
+	public function get_query_from_filter( $results, $query ) {
+		if ( ! empty( $query->query_vars['post__in'] ) && $query->query_vars['ignore_sticky_posts'] ) {
+			$this->sticky_query_vars = $query->query_vars;
+		}
+
+		return $results;
+	}
```

Comment:
I found out last week the test suite has a utility class to make this easier & avoid the property and filter function.

The test probably needs an `@covers` too.

```suggestion

	/**
	 * @ticket 36907
	 */
	public function test_stickies_nest_query() {
		$filter = new MockAction();
		add_filter( 'posts_pre_query', array( $filter, 'filter' ), 10, 2 );
		$this->go_to( '/' );
		$sticky_query_vars = $filter->get_args()[1][1]->query_vars;

		$q = $GLOBALS['wp_query'];
		$this->assertNotEmpty( $sticky_query_vars['posts_per_page'] );
		$this->assertSame( $q->query_vars['suppress_filters'], $sticky_query_vars['suppress_filters'] );
		$this->assertSame( $q->query_vars['update_post_meta_cache'], $sticky_query_vars['update_post_meta_cache'] );
		$this->assertSame( $q->query_vars['update_post_term_cache'], $sticky_query_vars['update_post_term_cache'] );
		$this->assertSame( $q->query_vars['lazy_load_term_meta'], $sticky_query_vars['lazy_load_term_meta'] );
		$this->assertSame( $q->query_vars['cache_results'], $sticky_query_vars['cache_results'] );
		$this->assertTrue( $sticky_query_vars['ignore_sticky_posts'] );
		$this->assertTrue( $sticky_query_vars['no_found_rows'] );
	}
```

Reply:
Do you think this improves the readability?

Reply:
A little, I guess 🤷 

It's nice to have less code and everything in one spot, but I agree that `[1][1]` to get the second argument of the second call isn't the clearest. Maybe chuck a comment in there.

Reply:
Fixed.